streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -708,6 +814,22 @@ private void connectStateStoreNameToSourceTopics(final String stateStoreName,
                 Collections.unmodifiableSet(sourceTopics));
     }
 
+
+    private <T> void maybeAddToResetList(Collection<T> earliestResets, Collection<T> latestResets, AutoOffsetReset offsetReset, T item) {
+        if (offsetReset != null) {
```

Comment:
I needed to put this check in otherwise we get a NPE. As an alternative should I create another enum NONE and use that under the covers instead of null?